ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java

```diff
@@ -390,7 +390,22 @@ public int getIntValue() throws IOException {
     // @since 2.17
     private int _getIntValue() throws IOException {
         try {
-            return _reader.intValue();
+            if (getNumberType() == NumberType.LONG) {
```

Comment:
Minor: for performance reasons, I recommend storing this value and using it in both branches. ion-java has to do some bounds checking on the values, and we should avoid doing that twice when possible.

Reply:
Ok will do. Thanks @tgregg

Reply:
was about to suggest same, +1